src/SessionSetup/RetrieveInstancesWidget.cpp

```diff
@@ -118,40 +122,47 @@ void RetrieveInstancesWidget::InitialLoad(const std::optional<Project>& remember
                 return;
               }
 
-              CHECK(widget->ui_->projectComboBox->count() == 0);
-
-              RetrieveInstances::LoadProjectsAndInstancesResult& data{loading_result.value()};
-
-              QString default_project_text =
-                  QString("Default Project (%1)").arg(data.default_project.display_name);
-              widget->ui_->projectComboBox->addItem(default_project_text);
-
-              QVector<Project>& projects{data.projects};
-              std::sort(projects.begin(), projects.end(), [](const Project& p1, const Project& p2) {
-                return p1.display_name.toLower() < p2.display_name.toLower();
-              });
-
-              for (const Project& project : data.projects) {
-                QString text = project.display_name;
-                if (project == data.default_project) {
-                  text.append(" (default)");
-                }
-                widget->ui_->projectComboBox->addItem(text, QVariant::fromValue(project));
+              widget->OnInitialLoadingReturnedSuccess(loading_result.value());
+            });
+}
 
-                if (project == data.project_of_instances) {
-                  // last added item
-                  widget->ui_->projectComboBox->setCurrentIndex(
-                      widget->ui_->projectComboBox->count() - 1);
-                }
-              }
+void RetrieveInstancesWidget::OnInitialLoadingReturnedSuccess(
+    LoadProjectsAndInstancesResult initial_load_result) {
+  CHECK(ui_->projectComboBox->count() == 0);
+
+  QString default_project_text =
+      QString("Default Project (%1)").arg(initial_load_result.default_project.display_name);
+  ui_->projectComboBox->addItem(default_project_text);
+
+  QVector<Project>& projects{initial_load_result.projects};
+  std::sort(projects.begin(), projects.end(),
+            [](const Project& p1, const Project& p2) { return p1.display_name < p2.display_name; });
+
+  for (const Project& project : initial_load_result.projects) {
+    QString text = project.display_name;
+    if (project == initial_load_result.default_project) {
+      text.append(" (default)");
+    }
+    ui_->projectComboBox->addItem(text, QVariant::fromValue(project));
+
+    // initial_load_result.instance, is the list for a specific project, which is
```

Comment:
Nit: Is this sentence correct? The comma throws me off, but also without, I'm not sure I can make sense of it... I think this should be `initial_load_result.instances (*plural*) is the list ...`? So this only contains instanced for the project returned as `initial_load_result.project_of_instances`?

Reply:
Yes and yes. I changed it